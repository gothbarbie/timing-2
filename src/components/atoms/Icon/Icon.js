import React from 'react'
import styled from 'styled-components'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

// NOTE! All icons used must be manually added at src/index.js

const IconStyle = styled(FontAwesomeIcon)`
  color: ${({ color, theme }) => color === 'white' && theme.colors.white};
  color: ${({ color, theme }) => color === 'gray' && theme.colors.gray};
`

const Icon = ({ color, icon, size }) => (
  <IconStyle color={color} icon={icon} size={size} />
)

export default Icon
