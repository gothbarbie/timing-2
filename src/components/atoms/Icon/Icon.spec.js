import React from 'react'
import { shallow } from 'enzyme'

import Icon from './Icon'

describe('generic/atoms/Icon', () => {
  it('renders', () => {
    const component = shallow(<Icon icon={['fa', 'times']} />)

    expect(component).toMatchSnapshot()
  })
})
