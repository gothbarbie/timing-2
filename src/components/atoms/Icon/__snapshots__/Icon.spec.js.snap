// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generic/atoms/Icon renders 1`] = `
<Styled(FontAwesomeIcon)
  icon={
    Array [
      "fa",
      "times",
    ]
  }
/>
`;
